support/langkit_support-adalog-generic_main_support.adb

```diff
@@ -88,10 +88,21 @@ package body Langkit_Support.Adalog.Generic_Main_Support is
    begin
       Put_Line ("Solving relation:");
       Print_Relation (Rel);
-      Solve
-        (Rel,
-         Solution_Callback'Unrestricted_Access,
-         (Cut_Dead_Branches => True));
+
+      case Kind is
+      when Symbolic =>
+         Put_Line ("... without optimizations:");
+         Solve (Rel, Solution_Callback'Access, (Cut_Dead_Branches => False));
+         New_Line;
+
+         Put_Line ("... cut dead branches:");
+         Solve (Rel, Solution_Callback'Access, (Cut_Dead_Branches => True));
+         New_Line;
```

Comment:
It seems that it would be good to at least check and show an "ERROR" or "WARNING" when we have diverging solutions in both cases, because for now it would be easy to miss it. WDYT ?

Reply:
Will do, thanks! We’ll change it to a real error (not one that can be baselined) once the current aliasing bug is fixed.